ibis/tests/sql/test_compiler.py

```diff
@@ -6,9 +6,13 @@
 import ibis
 import ibis.expr.api as api
 import ibis.expr.operations as ops
-from ibis import impala  # noqa: E402
+from ibis.backends import impala  # noqa: E402
+from ibis.backends.impala.compiler import (  # noqa: E402
```

Comment:
I think those may be related to the errors in the CI. Can you move the imports inside the functions when they are being used, and also create an issue and add a `TODO` to them? SQL backends shouldn't depend on the impala backend.

If you don't mind, probably better to do this change in a separate PR (also with the bigquery one). You'll have a small conflict to fix here later, but shouldn't be difficult to resolve.

Reply:
Sure. Would this be part of #2449 or a new issue?

Reply:
or more generally, are any new issues needed now or will it all be in #2449?

Reply:
Some more work is needed after #2449 is merged. The biggest part is to move the operation registry to the new base SQL backend. You can probably wait until that refactoring is done, things will be easy then. But feel free to apply this suggestion if yoy want to move things forward with this PR, that will take some time.